libs/server/controllers/friend/src/lib/friend.service.ts

```diff
@@ -0,0 +1,180 @@
+import { PrismaService } from '@bregenz-bewegt/server-prisma';
+import {
+  FriendSearchResult,
+  CreateFriendRequestDto,
+  FriendAdresseeResult,
+  FriendRequesteeResult,
+  RevokeFriendRequestDto,
+  AcceptFriendRequestDto,
+  RejectFriendRequestDto,
+  RemoveFriendDto,
+} from '@bregenz-bewegt/shared/types';
+import { Injectable } from '@nestjs/common';
+import { User, FriendRequest } from '@prisma/client';
+
+@Injectable()
+export class FriendService {
+  constructor(private prismaService: PrismaService) {}
+
+  async getFriends(userId: User['id']): Promise<User[]> {
+    const { friends } = await this.prismaService.user.findUnique({
+      where: { id: userId },
+      select: { friends: true },
+    });
+
+    return friends;
+  }
+
+  async searchUserFriendByUsername(
+    query: string,
+    userId: User['id']
+  ): Promise<FriendSearchResult[]> {
+    const maxSearchResults = 100;
+    const users = await this.prismaService.user.findMany({
+      where: {
+        username: { contains: query },
+        AND: { id: { not: userId } },
+      },
+      take: maxSearchResults,
+    });
+
+    const { friendRequests } = await this.prismaService.user.findUnique({
+      where: { id: userId },
+      select: { friendRequests: { where: { requestee: { id: userId } } } },
+    });
+
+    return <FriendSearchResult[]>users.map((user) => ({
```

Comment:
wenn ma nach freunden suacht kann ma o die freunde finda di ma scho hot --> entweder mir tuan in da suchergebnisse die Freunde die er hot excluden, oder mir zoagen döt a "X" oder an Hacken

Reply:
guate punkt